homeassistant/components/lcn/services.py

```diff
@@ -222,13 +222,14 @@ class SendKeys(LcnServiceCall):
     """Sends keys (which executes bound commands)."""
 
     schema = LcnServiceCall.schema.extend({
-        vol.Required(CONF_KEYS): cv.matches_regex(r'^([a-dA-D][1-8])+$'),
+        vol.Required(CONF_KEYS): vol.All(
+            vol.Upper, cv.matches_regex(r'^([a-dA-D][1-8])+$')),
```

Comment:
The regex still matches lowercase, should be updated to just uppercase then?

Reply:
In my opinion, it shouldn't depend on whether the value in the config is upper- or lowercase. It is my intend to allow both cases at this point.
However, the subsequent logic relies on uppercase.

Reply:
But since we convert everything ot uppercase, we can remove the lowercase chars from the regex?

Reply:
Yes, you‘re right. I got confused with the order of the validators.